mono/io-layer/handles.c

```diff
@@ -527,16 +526,8 @@ gpointer _wapi_handle_new_fd (WapiHandleType type, int fd,
 
 	MONO_TRACE (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER, "%s: Assigning new fd handle %d", __func__, fd);
 
-	/* Prevent file share entries racing with us, when the file
```

Comment:
the other place where it's locking doesn't access any handle related function, so the hypothesis that it "prevent file share entries racing with us" is IMO false.